core/src/main/java/com/valtech/aemsaas/core/services/configuration/BaseFilterConfiguration.java

```diff
@@ -0,0 +1,16 @@
+package com.valtech.aemsaas.core.services.configuration;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.caconfig.annotation.Configuration;
+import org.apache.sling.caconfig.annotation.Property;
+
+@Configuration(label = "Base Filter - Configuration", description = "Base Filter Configuration")
```

Comment:
let's add some more description, e.g. `Base Filters are added for all search requests done for the defined content tree.`